### ISSUE TITLE
[bug] wrong navigation route calculated

### ISSUE THREAD

**Issue by damjang** [osmandapp/Osmand#3634]
I find a bug in Osmand when calculating a route for navigation. I checked and there are no osm data problem so I suppose that there is a bug in Osmand.
I have: Osmand+2.5.4 on Samsung galaxy S2 - android 4.1.2 / offline routing /car (but also other types shows this bug) / map Italy_friuli-venezia-giulia_europe 2017-03-01

The location where you can see the bug is this:
http://www.openstreetmap.org/directions?engine=osrm_car&route=45.71650%2C13.72800%3B45.72060%2C13.72610#map=17/45.71855/13.72785
![osm_ok](https://cloud.githubusercontent.com/assets/1283609/24776749/7927a0a2-1b22-11e7-8c8e-ed74863e13c9.png)

On osm site the route is calculated correctly: 2 right turn and under the bridge to the highway.

In osmand the same start/finish location, but the navigation route is jumping directly from the bridge to the bottom road.
![screenshot_2017-04-06-23-20-17](https://cloud.githubusercontent.com/assets/1283609/24776771/8b74b3bc-1b22-11e7-8cf1-3faba92e5097.png)

Maybe the "problem" is that there are 2 points with nearly identical position, the first on the way on the bridge, the second on the way under the bridge. But the ways are not connected. I can remove the 2 incriminated points but think that the bug should be corrected in osmand.

Thank you
Damjan G.

**Comment by vshcherb** [comment]
Yes unfortunately this is a problem ( 2 points with nearly identical position) .
